### PR TITLE
Tai Nua Keyboard

### PR DESCRIPTION
**Pull request by rdai**
Here is my submission for the Tai Nua Keyboard (tdd). Please let me know if you have any questions, thank you!

### CODE REVIEW

**Comment by mcdurdin** [comment]
@LornaSIL: I've added you to this so you can be aware of this new keyboard coming in.

@rdai: This PR looks excellent at first glance. I'm reviewing the code and making sure it all builds. I think I will only have a few minor changes for you once I have completed that process.

File: tainua/source/tainua.kps [hunk 1]
```diff
@@ -0,0 +1,2 @@
...
...Site URL="www.surehope.net">www.surehope.net</WebSite></Info><Files><File><Name>..\..\..\..\keyboard\keyman\tainua.kvk</Name><Description>File tainua.kvk</Description><CopyLocation>0</CopyLocation><FileType>.kvk</FileType></File><File><Name>..\..\..\..\keyboard\keyman\tainua.kmx</Name><Description>Keyboard Tai Nua</Description><CopyLocation>0</CopyLocation><FileType>.kmx</FileType></File><File><Name>..\..\..\..\keyboard\keyman\tainua.ico</Name><Description>File tainua.ico</Description><CopyLocation>0</CopyLocation><FileType>.ico</FileType></File><File><Name>..\welcome\welcome.htm</Name><Description>File welcome.htm</Description><CopyLocation>0</CopyLocation><FileType>.htm</FileType></File><File><Name>..\welcome\tainua-1.jpg</Name><Description>File tainua-1.jpg</Description><CopyLocation>0</CopyLocation><FileType>.jpg</FileType></File><File><Name>..\welcome\tainua-2.jpg</Name><Description>File tainua-2.jpg</Description><CopyLocation>0</CopyLocation><FileType>.jpg</FileType></File><File><Name>..\..\..\..\..\..\..\..\..\..\temp\TaiNuaAUnicode.ttf</Name><Description>Font TaiNua Font </Description><CopyLocation>0</CopyLocation><FileType>.ttf</FileType></File></Files><Strings/></Package>
```
Comment (mcdurdin) [inline tainua/source/tainua.kps]: * The files referenced in the package source file are in another path. Open this file in the package editor and then, either: (a) edit this file in the source view in the Package Editor and fix the paths by hand, or (b) in the Files tab in the package editor, remove each file, and then re-add it from this source folder.
* I note also the .ttf file is missing from the PR.
* Finally, the welcome files should be in /source/welcome, not in /welcome.

File: tainua/tainua.kpj [hunk 2]
```diff
@@ -0,0 +1,98 @@
+<?xml version="1.0" encoding="utf-8"?>
+<KeymanDeveloperProject>
+  <Options>
+    <BuildPath></BuildPath>
```
Comment (mcdurdin) [inline tainua/tainua.kpj]: This needs to be set to "$PROJECTPATH\build"

**Comment by mcdurdin** [comment]
Looking pretty good now thank you, but one file, source\tainua-layout.js, is still missing. This file defines the touch layout.

**Comment by rdai** [comment]
Hi Marc and Lorna,

 

Thanks for catching these! I hope I have fixed everything you mentioned below. They are recommitted.

 

Ray

**Comment by mcdurdin** [comment]
I made a few minor changes to the .kps and .kpj to fix paths, so you should do:
    git fetch upstream
    git merge upstream/master
to integrate those changes into your source. Now this is looking great. I think the next step is to push these keyboards into the public downloads -- are you wanting both Desktop and Touch targets?

**Comment by rdai** [comment]
Thanks Marc!

 

I am sorry the paths were a mess, I decided to move things half way through. 

 

Sorry I am still learning Git so it looks like upstream is the main (yours) repository and you have already incorporated the changes and these steams are for me to update my local repository, is that correct?

I’ve done the steps.

 

Yes, I would like both Desktop and Touch targets. I have another keyboard I will be submitting sometime that will have two keyboards, one for desktop, one for touch.

 

Thank you.
Ray

**Comment by mcdurdin** [comment]
Yes, you are correct, I have merged the changes and those steps are to update your local repository and get it into sync with the published repository. I will work on getting the compiled keyboard into the online repository shortly.

**Pull request merged**